### ISSUE TITLE
startup time of runtime-wasmtime shim

### ISSUE THREAD

**Issue by sangeeta1998** [containerd/runwasi#704]
The startup time of wasm containers (with --runtime=io.containerd.wasmtime.v1 --platform=wasm) for the same container (once in native rust, and once as wasm) although the size of wasm image are much smaller than the native ones, the startup time with the wasmtime shim is much slower.

**Comment by Mossaka** [comment]
Do you have the startup time comparison between the native and wasm containers?

**Comment by sangeeta1998** [comment]
[startup_amd64.txt](https://github.com/user-attachments/files/17471997/startup_amd64.txt)

> Do you have the startup time comparison between the native and wasm containers?

**Comment by Mossaka** [comment]
Just curious - were you using the latest released Wasmtime shim, or using the shim built from the main branch? The latest released wasmtime shim had a bug that slows down the startup time by 1 second, which was fixed in main. I shall publish a new version for the wasmtime shim. Let me know though because that will make a big difference in startup time.

**Comment by sangeeta1998** [comment]
Yes, I built it from the main branch of runwasi, with something like : cargo build --release --target=x86_64-unknown-linux-gnu --target-dir=./target/ -p containerd-shim-wasmtime, but the shim I have is from 26 September.

**Comment by Mossaka** [comment]
Some updates: I am working on adding more benchmarks and traces to the core crate and the Wasmtime shim with a hope that we can gain more insights on the performance of the shim.

**Comment by giteewif** [comment]
@sangeeta1998  Could you tell how you get the execution time of the container execution, I'm curious, there may be the reason that the compilation time of the wasm file.

**Comment by Mossaka** [comment]
We've added / are adding stress test for the shim process to benchmark the startup time / CPU / memory and disk usages. 

#734 #830 

What we've got is with the hello world OCI image we used for testing, It takes about 5.4s to run 1000 containers with c8d in the loop, and 1.7s without c8d in the loop.
